cogs/TextFilter.py

```diff
@@ -0,0 +1,701 @@
+    """
+    for ig in ignored:
+        if ig[2] == 'channel':
+            details = self.bot.get_channel(int(ig[3]))
+        else:
+            details = self.bot.get_user(int(ig[3]))
+        if details is not None:
+            embed.add_field(name="Name & ID", value=details.mention + " " + str(details.id), inline=False)
+        else:
+            embed.add_field(name="ID", value=ig[3], inline=False)
+    return embed
+
+
+async def send_to_moderation_channels(self, message):
+    """
+    Send details about deleted message to mod channels
+    :param message: The message in question which is being deleted
+    """
+    if is_moderation_channel_available(self, message.guild.id):
+        channels = self.tf_database_manager.get_mod_channel(message.guild.id)
+        for each_chan in channels:
+            channel = self.bot.get_channel(id=int(each_chan[0]))
+            await channel.send(embed=build_moderation_deleted_embed(message))
+
+
+def build_moderation_channel_embed(ctx, channel, action):
+    """
+    Builds a moderation embed which display some information about the mod channel being created/removed
+    :param ctx: The discord context
+    :param channel: The channel to be created/removed
+    :param action: either "Added" or "Removed" to tell the user what happened to the mod channel
+    :return embed: The moderation embed to be sent to the user
+    """
+    embed = create_default_embed(ctx)
+    embed.title = "Koala Moderation - Mod Channel " + action
+    embed.add_field(name="Channel Name", value=channel.mention)
+    embed.add_field(name="Channel ID", value=channel.id)
+    return embed
+
+
+def build_word_list_embed(ctx, all_words, all_types, all_regex):
+    """
+    Builds the embed that is sent to list all the filtered words
+    :param ctx: The discord context
+    :param all_words: List of all the filtered words in the guild
+    :param all_types: List of all the corresponding filter types for the words in the guild
+    :param all_regex: List of all regex in the guild
+    :return embed with information about the deleted message:
+    """
+    embed = create_default_embed(ctx)
+    embed.title = "Koala Moderation - Filtered Words"
+    if not all_words and not all_types and not all_regex:
+        embed.add_field(name="No words found", value="For more help with using the Text Filter try k!help TextFilter")
+    else:
+        embed.add_field(name="Banned Words", value=all_words)
+        embed.add_field(name="Filter Types", value=all_types)
+        embed.add_field(name="Is Regex?", value=all_regex)
+    return embed
+
+
+def create_default_embed(ctx):
+    """
+    Creates a default embed that all embeds share
+    :param ctx: The discord context
+    :return embed with basic information which should be built upon:
+    """
+    embed = discord.Embed()
+    embed.colour = KOALA_GREEN
+    embed.set_footer(text=f"Guild ID: {ctx.guild.id}")
+    return embed
+
+
+def build_channel_list_embed(self, ctx, channels):
```

Comment:
If self is used, it should be in the specified class and instead use self.build_channel_list_embed(ctx, channels) to call this method